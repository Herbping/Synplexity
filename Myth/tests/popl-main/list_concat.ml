type nat =
  | Z
  | S of nat

type list =
  | Nil
  | Cons of nat * list

type llist =
  | LNil
  | LCons of list * llist

let rec append (l1:list) (l2:list) : list =
  match l1 with
  | Nil u -> l2
  | Cons t -> Cons (#1 t, append (#2 t) l2)
;;

let rec list_concat : llist -> list |>
  /\(LNil -> [],
     LCons ([], LNil) -> [],
     LCons ([0], LNil) -> [0],
     LCons ([0], LCons([0], LNil)) -> [0;0],
     LCons ([1], LNil) -> [1],
     LCons ([1], LCons([1], LNil)) -> [1;1]
     ) = ?
