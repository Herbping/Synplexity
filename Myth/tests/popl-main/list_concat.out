(* Generated Thu Oct  8 20:40:57 2015 *)
fix f1 (l1:llist) : list ->
  match l1 with
  | LNil u1 -> Nil
  | LCons t1 -> (append (#1 t1)) (f1 (#2 t1))
